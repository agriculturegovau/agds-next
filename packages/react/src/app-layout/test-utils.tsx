import { Fragment } from 'react';
import { Text } from '../text';
import { NotificationBadge } from '../notification-badge';
import { VisuallyHidden } from '../a11y';
import {
	ChartLineIcon,
	EmailIcon,
	ExitIcon,
	HelpIcon,
	HomeIcon,
	SuccessIcon,
	SettingsIcon,
	FactoryIcon,
	AvatarIcon,
	ChevronsLeftIcon,
	MinusIcon,
} from '../icon';
import {
	DropdownMenuDivider,
	DropdownMenuItem,
	DropdownMenuItemLink,
	DropdownMenuPanel,
	DropdownMenuGroup,
	DropdownMenuGroupLink,
	DropdownMenuItemRadio,
} from '../dropdown-menu';

export const navigationItems = (businessName: string) => [
	{
		options: { disableGroupPadding: true },
		items: [
			{
				label: 'Back to my account',
				icon: ChevronsLeftIcon,
				href: '/account',
			},
		],
	},
	{
		options: { disableGroupPadding: true },
		items: [
			{
				label: (
					<Fragment>
						<Text fontWeight="bold" fontSize="xs">
							{businessName}
						</Text>
						<Text color="muted" fontSize="xs">
							ABN: 00 000 000 000
						</Text>
					</Fragment>
				),
			},
		],
	},
	[
		{
			label: 'Dashboard',
			icon: HomeIcon,
			href: '/',
		},
		{
			label: 'Establishments',
			icon: FactoryIcon,
			href: '/establishments',
			items: [
				{
					label: 'Canberra',
					icon: MinusIcon,
					href: '/establishments/canberra',
				},
				{
					label: 'Sydney',
					icon: MinusIcon,
					href: '/establishments/sydney',
				},
			],
		},
		{
			label: 'Data and Insights',
			icon: ChartLineIcon,
			href: '/intelligence',
		},
		{
			label: 'Compliance',
			icon: SuccessIcon,
			href: '/compliance',
			items: [
				{
					label: 'Audit',
					icon: MinusIcon,
					href: '/compliance/audit',
				},
				{
					label: 'Certificates',
					icon: MinusIcon,
					href: '/compliance/certificates',
				},
			],
		},
	],

	[
		{
			label: 'Help',
			icon: HelpIcon,
			href: '/help',
		},
	],
	[
		{
			label: 'Sign out',
			onClick: console.log,
			icon: ExitIcon,
		},
	],
];

export const exampleData = {
	userNames: {
		short: 'Ab Cd',
		regular: 'Toto Wolff',
		medium: 'Benjamin Harrington',
		long: 'Benjamin Alexander Harrington-Smythe',
	},
	businessNames: {
		short: ['Lorem', 'Ipsum', 'Company'],
		regular: ['Antfix', 'Produce Fresh', 'Organic Co'],
		medium: [
			'Malesuada lacus a tortor blandit fermentum phasellus',
			'Iaculis tortor duis ante nec risus elementum id ui',
			'Purus tortor lacus malesuada phasellus ipsum ex duis libero ante id',
		],
		long: [
			'Duis quis iaculis libero nec consectetur luctus ex in malesuada lacus a tortor blandit fermentum phasellus iaculis ipsum nec purus aliquet id elementum risus duis ante ui',
			'Duis ut a libero duis nisl ex lacus in luctus viverra iaculis iaculis tortor duis malesuada blandit purus diam ipsum ante nec risus elementum id ui nec quis aliquet phasellus fermentum',
			'Blandit iaculis iaculis quis ante diam viverra elementum ui risus nec luctus purus tortor lacus malesuada phasellus ipsum ex duis libero ante id Duis fermentum nisl aliquet duis in a nec ut consectetur',
		],
	},
};

export function ExampleAccountDropdown({
	businesses = exampleData.businessNames.regular,
	onBusinessChange,
	selectedBusinessName,
}: {
	businesses?: string[];
	onBusinessChange: (value: string) => void;
	selectedBusinessName: string;
}) {
	return (
		<DropdownMenuPanel palette="light">
			<DropdownMenuGroup label="Businesses">
				{businesses.map((businessName) => (
					<DropdownMenuItemRadio
						key={businessName}
						checked={businessName === selectedBusinessName}
						secondaryText="ABN 00 000 000 000"
						onClick={() => onBusinessChange?.(businessName)}
					>
						{businessName}
					</DropdownMenuItemRadio>
				))}
				<DropdownMenuGroupLink href="#">View all</DropdownMenuGroupLink>
			</DropdownMenuGroup>
			<DropdownMenuDivider />
			<DropdownMenuGroup label="My account">
				<DropdownMenuItemLink href="/profile" icon={AvatarIcon}>
					Profile
				</DropdownMenuItemLink>
				<DropdownMenuItemLink
					href="/messages"
					icon={EmailIcon}
					endElement={
						<span>
							<NotificationBadge tone="action" value={6} max={99} aria-hidden />
							<VisuallyHidden>, 6 unread</VisuallyHidden>
						</span>
					}
				>
					Messages
				</DropdownMenuItemLink>
				<DropdownMenuItemLink href="/account-settings" icon={SettingsIcon}>
					Account settings
				</DropdownMenuItemLink>
			</DropdownMenuGroup>
			<DropdownMenuDivider />
			<DropdownMenuItem onClick={() => console.log('sign out')} icon={ExitIcon}>
				Sign out
			</DropdownMenuItem>
		</DropdownMenuPanel>
	);
}
