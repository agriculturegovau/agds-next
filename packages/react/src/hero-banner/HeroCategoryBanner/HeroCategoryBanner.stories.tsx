import { Meta, StoryObj } from '@storybook/react';
import { Box } from '../../box';
import {
	HeroCategoryBanner,
	HeroCategoryBannerProps,
} from './HeroCategoryBanner';
import {
	HeroCategoryBannerSubtitle,
	HeroCategoryBannerTitle,
} from './HeroCategoryBannerTitle';

type HeroCategoryBannerStoryProps = HeroCategoryBannerProps & {
	title: string;
	subtitle: string;
};

function Template({ title, subtitle, ...props }: HeroCategoryBannerStoryProps) {
	return (
		<HeroCategoryBanner {...props}>
			<HeroCategoryBannerTitle>{title}</HeroCategoryBannerTitle>
			<HeroCategoryBannerSubtitle>{subtitle}</HeroCategoryBannerSubtitle>
		</HeroCategoryBanner>
	);
}

const meta: Meta<HeroCategoryBannerStoryProps> = {
	title: 'layout/HeroBanner/HeroCategoryBanner',
	component: HeroCategoryBanner,
	parameters: {
		layout: 'fullscreen',
	},
	args: {
		title: 'Website hero banner title - xxl/display (H1)',
		subtitle: 'Short hero banner sentence - md/default (P)',
		image: (
			<img
				src="https://design-system.agriculture.gov.au/img/placeholder/hero-banner.jpeg"
				role="presentation"
				alt=""
			/>
		),
	},
	render: Template,
};

export default meta;

type Story = StoryObj<HeroCategoryBannerStoryProps>;

export const Basic: Story = {
	args: {},
};

export const OnBodyAlt: Story = {
	name: 'On bodyAlt background',
	args: {},
	render: (args) => (
		<Box paddingY={3} background="bodyAlt">
			<Template {...args} />
		</Box>
	),
};

export const WithoutImage: Story = {
	args: {
		image: undefined,
	},
};
