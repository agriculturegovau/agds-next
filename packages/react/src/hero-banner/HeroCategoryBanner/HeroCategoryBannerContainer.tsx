import { PropsWithChildren } from 'react';
import { Box } from '../../box';

export type HeroCategoryBannerContainerProps = PropsWithChildren<{}>;

export const HeroCategoryBannerContainer = ({
	children,
}: HeroCategoryBannerContainerProps) => {
	return (
		<Box
			as="section"
			background="contextAlt"
			css={{ position: 'relative', overflow: 'hidden' }}
		>
			{children}
		</Box>
	);
};
