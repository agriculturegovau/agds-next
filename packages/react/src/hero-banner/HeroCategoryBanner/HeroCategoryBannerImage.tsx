import { PropsWithChildren } from 'react';
import { Box } from '../../box';
import { boxPalette } from '../../core';
import { HeroBannerBackground, backgroundMap } from '../utils';

export type HeroCategoryBannerImageProps = PropsWithChildren<{
	background: HeroBannerBackground;
}>;

export const HeroCategoryBannerImage = ({
	children,
	background,
}: HeroCategoryBannerImageProps) => {
	const backgroundVar = backgroundMap[background];
	return (
		<Box
			css={{
				position: 'absolute',
				top: 0,
				right: 0,
				bottom: 0,

				'&::after': {
					content: '""',
					pointerEvents: 'none',
					position: 'absolute',
					inset: 0,
					background: `linear-gradient(90deg, ${boxPalette[backgroundVar]} 0px, rgba(255, 255, 255, 0.0) 360px)`,
				},

				img: {
					width: '100%',
					height: '100%',
					objectFit: 'cover',
					objectPosition: 'center',
				},
			}}
			display={['none', 'none', 'block']}
			width="40%"
		>
			{children}
		</Box>
	);
};
