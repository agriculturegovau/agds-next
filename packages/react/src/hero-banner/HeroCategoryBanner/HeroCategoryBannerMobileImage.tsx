import { ReactNode } from 'react';
import { Box } from '../../box';

export type HeroCategoryBannerMobileImageProps = { children: ReactNode };

export const HeroCategoryBannerMobileImage = ({
	children,
}: HeroCategoryBannerMobileImageProps) => (
	<Box
		css={{
			position: 'relative',
			overflow: 'hidden',
			paddingTop: '40%',
			'& img': {
				position: 'absolute',
				top: '50%',
				left: 0,
				right: 0,
				transform: 'translateY(-50%)',
				maxWidth: '100%',
			},
		}}
		display={{ xs: 'block', md: 'none' }}
	>
		{children}
	</Box>
);
