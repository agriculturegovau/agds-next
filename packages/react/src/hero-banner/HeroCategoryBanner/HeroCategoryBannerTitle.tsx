import { ReactNode } from 'react';
import { tokens } from '../../core';
import { Heading } from '../../heading';
import { Text } from '../../text';

export type HeroCategoryBannerTitleProps = { children: ReactNode };

export const HeroCategoryBannerTitle = ({
	children,
}: HeroCategoryBannerTitleProps) => (
	<Heading fontSize="xxl" type="h1">
		{children}
	</Heading>
);

export type HeroCategoryBannerSubTitleProps = { children: ReactNode };

export const HeroCategoryBannerSubtitle = ({
	children,
}: HeroCategoryBannerSubTitleProps) => (
	<Text as="p" fontSize="md" maxWidth={tokens.maxWidth.bodyText}>
		{children}
	</Text>
);
