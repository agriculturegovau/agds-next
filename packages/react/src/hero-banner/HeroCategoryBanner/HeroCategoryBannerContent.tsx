import { PropsWithChildren, ReactNode } from 'react';
import { tokens } from '../../core';
import { Flex } from '../../flex';
import { Stack } from '../../stack';
import { Content } from '../../content';
import { HeroBannerBackground } from '../utils';
import { HeroCategoryBannerImage } from './HeroCategoryBannerImage';

export type HeroCategoryBannerContentProps = PropsWithChildren<{
	image?: ReactNode;
	background: HeroBannerBackground;
}>;

export const HeroCategoryBannerContent = ({
	children,
	image,
	background,
}: HeroCategoryBannerContentProps) => {
	return (
		<Content>
			<Flex>
				<Stack
					gap={1.5}
					width={['100%', '100%', image ? '60%' : '100%']}
					paddingTop={{ xs: 2, md: 4 }}
					paddingBottom={{ xs: 3, md: 4 }}
					css={{ zIndex: tokens.zIndex.elevated }}
				>
					{children}
				</Stack>
				{image ? (
					<HeroCategoryBannerImage background={background}>
						{image}
					</HeroCategoryBannerImage>
				) : null}
			</Flex>
		</Content>
	);
};
