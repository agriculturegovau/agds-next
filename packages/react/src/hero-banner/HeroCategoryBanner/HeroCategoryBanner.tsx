import { PropsWithChildren, ReactNode } from 'react';
import { HeroBannerBackground } from '../utils';
import { HeroCategoryBannerContent } from './HeroCategoryBannerContent';
import { HeroCategoryBannerMobileImage } from './HeroCategoryBannerMobileImage';
import { HeroCategoryBannerContainer } from './HeroCategoryBannerContainer';

export type HeroCategoryBannerProps = PropsWithChildren<{
	/** The hero image */
	image?: ReactNode;
	/** The background of the component */
	background?: HeroBannerBackground;
}>;

export const HeroCategoryBanner = ({
	children,
	image,
	background = 'bodyAlt',
}: HeroCategoryBannerProps) => (
	<HeroCategoryBannerContainer background={background}>
		{image ? (
			<HeroCategoryBannerMobileImage>{image}</HeroCategoryBannerMobileImage>
		) : null}
		<HeroCategoryBannerContent background={background} image={image}>
			{children}
		</HeroCategoryBannerContent>
	</HeroCategoryBannerContainer>
);
