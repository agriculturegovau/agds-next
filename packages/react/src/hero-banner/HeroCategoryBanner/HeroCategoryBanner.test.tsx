import '@testing-library/jest-dom';
import 'html-validate/jest';
import { axe, toHaveNoViolations } from 'jest-axe';
import { cleanup, render } from '../../../../../test-utils';
import { HeroCategoryBanner } from './HeroCategoryBanner';
import {
	HeroCategoryBannerTitle,
	HeroCategoryBannerSubtitle,
} from './HeroCategoryBannerTitle';

expect.extend(toHaveNoViolations);

afterEach(cleanup);

function renderHeroCategoryBanner() {
	return render(
		<HeroCategoryBanner
			image={
				<img
					src="/img/placeholder/hero-banner.jpeg"
					role="presentation"
					alt=""
				/>
			}
		>
			<HeroCategoryBannerTitle>
				Website hero banner title
			</HeroCategoryBannerTitle>
			<HeroCategoryBannerSubtitle>
				Hero banner sub title
			</HeroCategoryBannerSubtitle>
		</HeroCategoryBanner>
	);
}

describe('HeroCategoryBanner', () => {
	it('renders correctly', () => {
		const { container } = renderHeroCategoryBanner();
		expect(container).toMatchSnapshot();
	});

	it('renders valid HTML with no a11y violations', async () => {
		const { container } = renderHeroCategoryBanner();
		expect(container).toHTMLValidate({
			extends: ['html-validate:recommended'],
		});
		expect(await axe(container)).toHaveNoViolations();
	});
});
