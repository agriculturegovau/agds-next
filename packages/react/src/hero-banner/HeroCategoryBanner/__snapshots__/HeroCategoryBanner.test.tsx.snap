// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroCategoryBanner renders correctly 1`] = `
<div>
  <section
    class="css-1p0zx3h-boxStyles-HeroCategoryBannerContainer"
  >
    <div
      class="css-1g5msad-boxStyles-HeroCategoryBannerMobileImage"
    >
      <img
        alt=""
        role="presentation"
        src="/img/placeholder/hero-banner.jpeg"
      />
    </div>
    <div
      class="css-ou39vu-boxStyles"
    >
      <div
        class="css-1rhmzip-boxStyles"
      >
        <div
          class="css-1jhrbpg-boxStyles"
        >
          <div
            class="css-shh8uh-boxStyles-HeroCategoryBannerContent"
          >
            <h1
              class="css-18pa81a-boxStyles"
            >
              Website hero banner title
            </h1>
            <p
              class="css-kvrtyp-boxStyles-Text"
            >
              Hero banner sub title
            </p>
          </div>
          <div
            class="css-uhwge9-boxStyles-HeroCategoryBannerImage"
          >
            <img
              alt=""
              role="presentation"
              src="/img/placeholder/hero-banner.jpeg"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
